### PR TITLE
Provide `intellijVersions` by `rootProject.extra` instead of an extension

### PR DESCRIPTION
**Pull request by PawelLipski**
image

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -13,11 +13,6 @@ plugins {
   scala
 }
 
-project.extensions.add(
```
Comment (PawelLipski) [inline build.gradle.kts:16]: WDYT @MaciejG604? I don't think using an extension is justified/necessary here, `intellijVersions` don't provide any methods, just read-only properties

File: buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/UpdateIntellijVersions.kt [hunk 2]
```diff
@@ -6,8 +6,8 @@ import com.virtuslab.gitmachete.buildsrc.IntellijVersionHelper.toBuildNumber
 import com.virtuslab.gitmachete.buildsrc.IntellijVersionHelper.versionIsNewerThan
 import org.gradle.api.DefaultTask
 import org.gradle.api.tasks.TaskAction
-import org.gradle.kotlin.dsl.get
-import org.gradle.kotlin.dsl.getByType
+import org.gradle.kotlin.dsl.extra
+import org.gradle.kotlin.dsl.provideDelegate
```
Comment (PawelLipski) [inline buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/UpdateIntellijVersions.kt:10]: `extra` don't provide `getValue`/`setValue` methods [required for a delegate](https://kotlinlang.org/docs/delegated-properties.html#property-delegate-requirements) out of themselves, hence this extra magic is needed

**Comment by PawelLipski** [comment]
Spurious UI test failure, yet unknown one:

image

Opened #1210 for that...

File: build.gradle.kts (hunk 1 above)
Comment (MaciejG604) [reply build.gradle.kts:16]: Yeah, that is true.

**Review by MaciejG604 — APPROVED** [review]

**Pull request merged**